src/constants.ts

```diff
@@ -5,6 +5,8 @@ import { BigNumber } from '@ethersproject/bignumber';
 
 import type { SimulationRoles } from './types';
 
+export const PRECISION = '100000000000000000';
```

Comment:
I believe this should be 10^18 instead of 10^17 to ensure higher accuracy (honey has 18 decimals, and its mint rate is in 18 decimals precision too `998000000000000000`)

Reply:
actually the precision is just being used to ensure there are no losses to rounding until the calculations are done. in this case the amount doesn't actually have to match the token's decimal precision. however, i did intend to use 10^18 and just miscounted. i'll update it